pennylane/qnodes/base.py

```diff
@@ -689,15 +722,7 @@ def _default_args(self, kwargs):
                     continue  # unknown parameter, but **kwargs will take it TODO should it?
                 raise QuantumFunctionError("Unknown quantum function parameter '{}'.".format(name))
 
-            default_parameter = s.par.default
-
-            # The following is a check of the default parameter which works for numpy
-            # arrays as well (if it is a numpy array, each element is checked separately).
-            # FIXME why are numpy array default values not good automatically?
-            correct_default_parameter = any(d == inspect.Parameter.empty for d in default_parameter)\
-                                        if isinstance(default_parameter, np.ndarray)\
-                                        else default_parameter == inspect.Parameter.empty
-            if s.par.kind in forbidden_kinds or correct_default_parameter:
+            if s.par.kind in forbidden_kinds or s.par.default is inspect.Parameter.empty:
```

Comment:
The issue here was solved by using `is` instead of `==`, since `inspect.Parameter.empty` is a singleton object.

Reply:
Nice!